libraries/AP_Notify/AP_Notify.cpp

```diff
@@ -199,6 +199,76 @@ void AP_Notify::add_backend_helper(NotifyDevice *backend)
 
 #define ADD_BACKEND(backend) do { add_backend_helper(backend); if (_num_devices >= CONFIG_NOTIFY_DEVICES_MAX) return;} while(0)
 
+void AP_Notify::set_nav_lights(void)
+{
+  AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev1 =  std::move(hal.i2c_mgr->get_device(0, 0x51,100000, true, 20));
+
+  bool dev1sem = false;
+  bool dev2sem = false;
+  bool dev3sem = false;
+  bool dev4sem = false;
+
+  if(!dev1->get_semaphore()->take(10))
+    gcs().send_text(MAV_SEVERITY_CRITICAL, "dev1 sem take fail");
+  else{
+    gcs().send_text(MAV_SEVERITY_CRITICAL, "dev1 sem take good");
+    dev1sem = true;
+  }
+
+  dev1->write_register(0x0a,0x40);
+  dev1->write_register(0x0B,0x01);
+
+  if(dev1sem)
+    dev1->get_semaphore()->give();
+
+
+  AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev2 =  std::move(hal.i2c_mgr->get_device(0, 0x52,100000, true, 20));
```

Comment:
Do `OwnPtr`s need to be destroyed after use?

Reply:
Not sure, basically it's a px4 compatible version of std::unique_ptr. Would a std::unique_ptr need to be destroyed?

Reply:
Per our discussion, it should be fine